modules/core/02-client/keeper/params.go

```diff
@@ -6,19 +6,28 @@
 	"github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
 )
 
-// GetAllowedClients retrieves the allowed clients from the paramstore
+// GetAllowedClients retrieves the allowed clients from the params stored.
 func (k Keeper) GetAllowedClients(ctx sdk.Context) []string {
-	var res []string
-	k.paramSpace.Get(ctx, types.KeyAllowedClients, &res)
-	return res
+	p := k.GetParams(ctx)
+	return p.AllowedClients
 }
 
 // GetParams returns the total set of ibc-client parameters.
 func (k Keeper) GetParams(ctx sdk.Context) types.Params {
-	return types.NewParams(k.GetAllowedClients(ctx)...)
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get([]byte(types.ParamsKey))
+	if bz == nil {
+		panic("ibc client params are not set in store")
```

Comment:
This is called in BeginBlocker when updating the localhost client. Personally I'd rather avoid panics entirely in these situations.

Reply:
Gotcha, will switch to err. Out of curiosity, what's the difference in semantics between handling an error/panic in begin blocker?

Reply:
Is empty params what the sdk does? I know @colin-axner mentioned briefly that he had a minor preference for panicking, but it would be nice if we could avoid it imo. 

For context, panicking in the BeginBlocker abci call is not caught by sdk's baseapp. Panicking in msg server handlers is "fine" because its caught by the grpc recovery interceptor, but doing so in BeginBlocker would crash nodes and ultimately halt a chain

Reply:
Ah, good point, @damiannolan! I believe the [SDK does not panic](https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/x/mint/keeper/keeper.go#L100).

Reply:
My view is that unset params are a sign of a misconfigured chain which should be halted. It seems unsafe to process any transactions if certain parameters on chain haven't been decided on. My recommendation would be defense in depth, as we may choose to add params in the future which might not have a similar sane default value as allowed clients